server/index.js

```diff
@@ -56,6 +56,99 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/carts', (req, res, next) => {
+  // const sql = `
+  // SELECT    "cartId", "createdAt"
+  // FROM      "carts"
+  // `;
+  // db.query(sql)
+  //   .then(result => res.json(result.rows))
+  //   .catch(err => next(err));
+  if (req.session.cartId) {
+    const sql = `
+    select "c"."cartItemId",
+       "c"."price",
+       "p"."productId",
+       "p"."image",
+       "p"."name",
+       "p"."shortDescription"
+    from "cartItems" as "c"
+    join "products" as "p" using ("productId")
+    where "c"."cartId" = $1
+    `;
+    const param = [req.session.cartId];
+    db.query(sql, param)
+      .then(result => res.json(result.rows))
+      .catch(err => next(err));
+  } else {
+    return [];
+  }
+});
+
+app.post('/api/carts', (req, res, next) => {
+  const cart = req.body;
+  if (!parseInt(cart.productId, 10) || parseInt(cart.productId, 10) < 0) {
+    return res.status(400).json({
+      error: '"productId" must be a positive integer'
+    });
+  }
+  const sql1 = `
+  SELECT      "price"
+  FROM        "products"
+  WHERE       "productId" = $1
+  `;
+  const params1 = [cart.productId];
+  db.query(sql1, params1)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        res.status(400).json({
+          error: `"productId" ${cart.productId} is not found`
+        });
```

Comment:
You need to `throw` here to abort the promise chain. Responding doesn't cancel the promise chain.